src/FrameBase.re

```diff
@@ -48,6 +60,33 @@ external waitForXPath :
   Js.Promise.t(ElementHandle.t) =
   "";
 
+/* TODP: Add support [, ...args] */
 [@bs.send.pipe : t]
 external selectOneEval : (string, unit => unit) => Js.Promise.t('a) =
   "$eval";
+
+/* TODP: Add support [, ...args] */
+[@bs.send.pipe : t]
+external selectAllEval : (string, unit => unit) => Js.Promise.t('a) =
+  "$$eval";
+
+[@bs.send.pipe : t]
+external addScriptTag : tagOptions => Js.Promise.t(ElementHandle.t) =
+  "";
+
+
+[@bs.send.pipe : t]
+external addStyleTag : tagOptions => Js.Promise.t(ElementHandle.t) =
+  "";
+
+[@bs.send.pipe : t]
+external evaluate :
+  (unit => Js.Promise.t(Js.Json.t), [@bs.splice] array({..})) =>
```

Comment:
This will only ever work for functions taking no arguments, and the second parameter array can only ever be empty. Same for `evaluateHandle`. There are some ways around this we can experiment with, but I'd like to get this merged and we can iterate on it a bit.

Reply:
Let me put TODO comment on it and improve it later.

Reply:
Comment added in 067382b